src/main/java/org/opensearch/ad/AnomalyDetectorRunner.java

```diff
@@ -136,14 +144,17 @@ private void onFailure(Exception e, ActionListener<List<AnomalyResult>> listener
         // TODO return exception like IllegalArgumentException to explain data is not enough for preview
         // This also requires front-end change to handle error message correspondingly
         // We return empty list for now to avoid breaking front-end
+        if (e instanceof OpenSearchSecurityException) {
+            listener.onFailure(e);
+        }
```

Comment:
If exception is `OpenSearchSecurityException`, will call `onFailure` first, then call `onResponse`, it will throw connection closed exception.

Add `return` in this `if` block, or put `listener.onResponse(Collections.emptyList());` in `else` block.

Reply:
yeah, fixed this one and another similar bug.